extensions/blocks/podcast-episodes/edit.js

```diff
@@ -3,49 +3,126 @@
 /**
  * External dependencies
  */
+import { Component } from '@wordpress/element';
+import {
+	Button,
+	Disabled,
+	ExternalLink,
+	PanelBody,
+	Placeholder,
+	RangeControl,
+	TextControl,
+	Toolbar,
+} from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
-import { BlockIcon } from '@wordpress/block-editor';
-import { Placeholder, withNotices } from '@wordpress/components';
-import { useState } from '@wordpress/element';
+import { BlockControls, BlockIcon, InspectorControls } from '@wordpress/block-editor';
+import ServerSideRender from '@wordpress/server-side-render';
 
 /**
  * Internal dependencies
  */
 import './editor.scss';
-import icon from './icon';
-
-function PodcastEpisodesEdit( {
-	attributes,
-	className,
-	noticeOperations,
-	noticeUI,
-	setAttributes,
-} ) {
-	/**
-	 * Write the block editor UI.
-	 *
-	 * @returns {object} The UI displayed when user edits this block.
-	 */
-	const [ notice, setNotice ] = useState();
-
-	/* Call this function when you want to show an error in the placeholder. */
-	const setErrorNotice = () => {
-		noticeOperations.removeAllNotices();
-		noticeOperations.createErrorNotice( __( 'Put error message here.', 'jetpack' ) );
-	};
-
-	return (
-		<div className={ className }>
-			<Placeholder
-				label={ __( 'Podcast Episodes', 'jetpack' ) }
-				instructions={ __( 'Instructions go here.', 'jetpack' ) }
-				icon={ <BlockIcon icon={ icon } /> }
-				notices={ noticeUI }
-			>
-				{ __( 'User input goes here?', 'jetpack' ) }
-			</Placeholder>
-		</div>
-	);
+import { edit, queueMusic } from './icons/';
+
+const DEFAULT_MIN_ITEMS = 1;
+const DEFAULT_MAX_ITEMS = 10;
+
+class PodcastEpisodesEdit extends Component {
+	constructor() {
+		super( ...arguments );
+		this.toggleAttribute = this.toggleAttribute.bind( this );
+		this.onSubmitURL = this.onSubmitURL.bind( this );
+	}
+
+	toggleAttribute( propName ) {
+		return () => {
+			const value = this.props.attributes[ propName ];
+			const { setAttributes } = this.props;
+
+			setAttributes( { [ propName ]: ! value } );
+		};
+	}
+
+	onSubmitURL( event ) {
+		event.preventDefault();
+	}
+
+	render() {
+		const { url, itemsToShow } = this.props.attributes;
+		const { attributes, setAttributes } = this.props;
+
+		if ( ! url ) {
+			return (
+				<Placeholder
+					icon={ <BlockIcon icon={ queueMusic } /> }
+					label={ __( 'Podcast Episodes', 'jetpack' ) }
+					instructions={ __( 'Paste a link to your Podcast RSS feed.', 'jetpack' ) }
+				>
+					<form onSubmit={ this.onSubmitURL }>
+						<TextControl
+							placeholder={ __( 'Enter URL here…', 'jetpack' ) }
+							value={ url || '' }
+							onChange={ value => setAttributes( { url: value } ) }
```

Comment:
Instead of updating the attribute right from there, what would you think about doing it on form submit, like Pinterest does it?
https://github.com/Automattic/jetpack/blob/5fcf11693c5b59c8124cbea96f4a74593e1cccba/extensions/blocks/pinterest/edit.js#L105-L121

It may allow us to do some necessary validation before we save just about any string as url attribute.

Reply:
Yes, this needs to be updated I think. Otherwise typing every character would update a state and cause an item in the undo stack. That's not how other blocks work

Reply:
Actually, the core RSS block does it the same way. Maybe we can keep it as is because our block is essentially a copy of that with an enhanced frontend? https://github.com/WordPress/gutenberg/blob/3848806ccc7b8a65778366e80af7ea63258538d4/packages/block-library/src/rss/edit.js#L77

Reply:
Yeah, current code generates this class which is 100% not intended: 

```
class="wp-block-https://distributedbloglive.files.wordpress.com/2020/01/distributed-episode-18-with-jason-fried-1.mp3 wp-playlist wp-audio-playlist wp-playlist-light"
```

Reply:
I also noticed this and had the above pattern in my head. I also noticed the RSS block.

I'm inclined to go with the Pinterest flow as it avoids relying on the `editing` flag in state to determine what the block looks like and avoids us computing `state` from `props`.

Reply:
Updated to what @jeherve suggested